### PR TITLE
Documentation pushed to GitHub pages

### PR DESCRIPTION
**Pull request by solvip**
# What?

On every master build, all of our documentation is pushed to our GitHub pages page at https://hundarogborn.github.io/TicTacToe

To do this, I implemented a `bin/build_doc` which is responsible for building a Jekyll site and populating it with javadoc, coverage results etc.

I also implemented a travis script that orchestrates all this
# Why?

So that we can easily access pretty documentation from outside of the repo/build process
(also - for points)
# Reviewer checklist

[ ] - The change introduces no untested code
[ ] - The change passes all existing tests
[ ] - The change introduces no typos or formatting errors

### CODE REVIEW

File: bin/build [hunk 1]
```diff
@@ -23,12 +23,6 @@ if [ $? -ne 0 ]; then
     exit 1;
 fi
 
-echo "Running code coverage tests..."
-"$SCRIPT_DIR/../gradlew" jacocoTestReport
-if [ $? -ne 0 ]; then
-    echoerr "Code coverage tests failed - continue ..."
-fi
-
```
Comment (sveinng) [inline bin/build:31]: Why drop code coverage testing?

**Review by eirikura — APPROVED** [review]
Nice job :)

**Review by sveinng — COMMENTED** [review]
This is so sexy!!! 💯

**Review by rebekka123 — APPROVED** [review]
Super cool

File: bin/build (hunk 1 above)
Comment (solvip) [reply bin/build:31]: Ooops.  Cut instead of copy.  I'll reintroduce.

**Comment by sveinng** [comment]
Merge please :)

**Pull request merged**